### PR TITLE
Wayland egl

### PR DESCRIPTION
**Pull request by nongio**

### CODE REVIEW

File: Headers/wayland/WaylandServer.h [hunk 1]
```diff
@@ -132,7 +139,7 @@ struct window {
     struct xdg_positioner *positioner;
     struct zwlr_layer_surface_v1 *layer_surface;
     struct output *output;
-    WaylandCairoSurface *wcs;
+    _CAIRO_SURFACE_CLASSNAME *wcs;
```
Comment (fredkiefer) [inline Headers/wayland/WaylandServer.h:142]: Could this variable just be a CairoSurface? I don't see any specific method you are using on it after the creation. But I might be wrong here as the PR only shows the changed lines.

File: Source/cairo/WaylandCairoEglSurface.m [hunk 2]
```diff
@@ -0,0 +1,350 @@
...
+
+    int x = NSMinX(rect);
+    int y = NSMinY(rect);
+    int width = NSWidth(rect);
+    int height = NSHeight(rect);
+    window->buffer_needs_attach = YES;
+    if (window->configured) {
+        // if the window is configured, it's time to display the content on screen
+        // we are creating a window surface
+        if(_eglSurface == NULL) {
```
Comment (fredkiefer) [inline Source/cairo/WaylandCairoEglSurface.m:311]: Again this does not follow the GNUstep coding guidelines.

Comment (nongio) [reply Source/cairo/WaylandCairoEglSurface.m:311]: thanks for reviewing it, i planned to refactor the code after rebasing with the other PR

File: Headers/wayland/WaylandServer.h (hunk 1 above)
Comment (nongio) [reply Headers/wayland/WaylandServer.h:142]: good suggestion,
I closed this PR and made a new one, rebasing the code on the wayland branch. this change is applied there

File: Source/cairo/WaylandCairoEglSurface.m (hunk 2 above)
Comment (nongio) [reply Source/cairo/WaylandCairoEglSurface.m:311]: in here the refactored code following the guidelines
https://github.com/nongio/libs-back/pull/2

**Pull request merged**